### DISCUSSION TITLE
[FEAT]: Add support for bun:test to angular test integrations

### DISCUSSION THREAD

**Discussion by NateRadebaugh** [analogjs/analog#1281] [Ideas]
### Which scope/s are relevant/related to the feature request?

create-analog

### Information

Vitest is great, but bun:test is way faster.
> https://bun.sh/docs/cli/test
> https://bun.sh/docs/test/lifecycle
> https://x.com/jarredsumner/status/1632545090160975873

Is it feasible for the angular vitest integration to also work in a "bun test" context? I understand that test runner would not use vite under the hood and would likely need some translation step using some setup.ts file.

```bash
bun test --preload ./setup.ts
```

or

```toml
[test]
preload = ["./setup.ts"]
```

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

**Comment by brandonroberts** [comment]
We did something similar with an esbuild plugin

https://github.com/analogjs/analog/blob/beta/packages%2Fvite-plugin-angular%2Fesbuild.ts#L5

Maybe that implementation could be adapted for bun

**Comment by NateRadebaugh** [comment]
I've created a separate bug in bun's repo:
https://github.com/oven-sh/bun/issues/15240

I'm not sure whether this belongs in analog or bun's code, but bun does support plugins and I think may require some custom angular-specific integrations to get the angular parser to run on external template HTML files
> https://bun.sh/docs/bundler/plugins
